enclave-runtime/src/utils.rs

```diff
@@ -171,3 +172,28 @@ pub(crate) fn get_stf_executor_from_solo_or_parachain() -> Result<Arc<EnclaveStf
 		};
 	Ok(stf_executor)
 }
+
+pub(crate) fn get_stf_enclave_signer_from_solo_or_parachain() -> Result<Arc<EnclaveStfEnclaveSigner>>
+{
+	let stf_enclave_signer =
+		if let Ok(solochain_handler) = GLOBAL_INTEGRITEE_SOLOCHAIN_HANDLER_COMPONENT.get() {
+			match &*solochain_handler.import_dispatcher {
+				BlockImportDispatcher::TriggeredDispatcher(dispatcher) =>
+					dispatcher.block_importer.indirect_calls_executor.stf_enclave_signer.clone(),
```

Comment:
very deep nesting here. I can live with this for now, but this smells suboptimal

Reply:
Yes, no worth refactoring this specifically, as I hope to remove this systematically by changing the underlying structure of the globals.